src/main/java/reposense/commits/CommitResultAggregator.java

```diff
@@ -27,14 +27,14 @@ public class CommitResultAggregator {
      * Returns the {@link CommitContributionSummary} generated from aggregating the {@code commitResults}.
      * Uses {@code config} to obtain details like author name, since date and timezone.
      */
-    public static CommitContributionSummary aggregateCommitResults(
-            RepoConfiguration config, List<CommitResult> commitResults) {
+    public CommitContributionSummary aggregateCommitResults(
+            RepoConfiguration config, List<CommitResult> commitResults, ReportGenerator reportGenerator) {
```

Comment:
Passing in `ReportGenerator` as an argument here as well as to `generateCommitSummary` is in my opinion an acceptable way as its the most direct way with the least hassle since it can no longer use `ReportGenerator`'s static method.